HopsanGUI/Utilities/HighlightingUtilities.cpp

```diff
@@ -348,60 +354,74 @@ CppHighlighter::CppHighlighter(QTextDocument *parent)
                     << "\\bslots\\b" << "\\bstatic\\b" << "\\bstruct\\b"
                     << "\\btemplate\\b" << "\\btypedef\\b" << "\\btypename\\b"
                     << "\\bunion\\b" << "\\bunsigned\\b" << "\\bvirtual\\b"
-                    << "\\bvoid\\b" << "\\bvolatile\\b" << "\\busing\\b";
-    foreach (const QString &pattern, keywordPatterns) {
+                    << "\\bvoid\\b" << "\\bvolatile\\b" << "\\busing\\b"
+                    << "\\bbool\\b" << "\\bif\\b" << "\\belse\\b"
+                    << "\\bfor\\b" << "\\bforeach\\b" << "\\bwhile\\b"
+                    << "\\bswitch\\b" << "\\bcase\\b" << "\\bdefault\\b"
+                    << "\\bbreak\\b" << "\\breturn\\b" << "\\bif\\()";
+    foreach (const QString &pattern, keywordPatterns)
+    {
         rule.pattern = QRegExp(pattern);
-        rule.format = keywordFormat;
-        highlightingRules.append(rule);
+        rule.format = mKeywordFormat;
+        mHighlightingRules.append(rule);
+    }
+
+    mHopsanKeywordFormat.setForeground(Qt::darkMagenta);
+    mHopsanKeywordFormat.setFontWeight(QFont::Normal);
+    QStringList hopsanKeywordPatterns;
+    hopsanKeywordPatterns << "\\bPort\\b" << "\\bFirstOrderTransferFunctionVariable\\b" << "\\bSecondOrderTransferFunctionVariable\\b"
+                          << "\\bFirstOrderTransferFunction\\b" << "\\bSecondOrderTransferFunction\\b"
```

Comment:
This list should be longer, can say right now what is missing. Maybe we should use external text file with keywords to make it easier to add new ones.

Reply:
I added all component utilities. There are probably more things we can add later.